visitor/TreeVisitor.cpp

```diff
@@ -56,22 +56,22 @@ bool TreeVisitor::VisitEnumDecl(clang::EnumDecl *enumdecl){
     //Replace "enum x" with enum_x in scala
     typeTranslator.AddTranslation("enum " + name, "enum_" + name);
 
-    if(name == "" && enumdecl->getTypedefNameForAnonDecl()){
+    if(name.empty() && enumdecl->getTypedefNameForAnonDecl()){
         name = enumdecl->getTypedefNameForAnonDecl()->getNameAsString();
     }
 
-    if(name != ""){
-     declarations += "\ttype enum_" + name + " = native.CInt\n";
+    if(!name.empty()){
+        ir->getLibObject().addType("type enum_" + name + " = native.CInt");
     }
 
     int i = 0;
     for (const clang::EnumConstantDecl* en : enumdecl->enumerators()){
         std::string ename = handleReservedWords(en->getNameAsString());
 
-        if(name != ""){
-            enums += "\tfinal val enum_" + name + "_" + ename + " = " + std::to_string(i++) + "\n";
+        if(!name.empty()){
+            ir->getEnumsObject().addEnum("final val enum_" + name + "_" + ename + " = " + std::to_string(i++));
```

Comment:
Enums may have explicit value, so  a counter won’t work. Digging into the API something like the following might work:

```c
en->getInitVal()->getLimitedValue()
```

If you prefer not to fix it in this PR, we should create a ticket.